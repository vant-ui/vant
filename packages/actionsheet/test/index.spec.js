import { mount } from '@vue/test-utils';
import Actionsheet from '../';
import demoTest from '../../../test/demo-test';

demoTest('actionsheet');

describe('Actionsheet', () => {
  it('Callback events', () => {
    const callback = jest.fn();
    const wrapper = mount(Actionsheet, {
      propsData: {
        actions: [
          { name: 'Option', callback },
          { name: 'Option' }
        ],
        cancelText: 'Cancel'
      }
    });

    wrapper.findAll('li').trigger('click');
    wrapper.find('.van-actionsheet__cancel').trigger('click');

    expect(callback.mock.calls.length).toBe(1);
    expect(wrapper.emitted('cancel')).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
