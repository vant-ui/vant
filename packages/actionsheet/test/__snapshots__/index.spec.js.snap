// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actionsheet Callback events 1`] = `
<div class="van-actionsheet" style="display: none;" name="van-slide-bottom">
  <ul class="van-hairline--bottom">
    <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">Option</span>
      <!---->
    </li>
    <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">Option</span>
      <!---->
    </li>
  </ul>
  <div class="van-actionsheet__cancel van-hairline--top">Cancel</div>
</div>
`;

exports[`renders actionsheet correctly 1`] = `
<demo-section>
  <demo-block title="基础用法">
    <button class="van-button van-button--default van-button--normal">
      <!----><span class="van-button__text"></span></button>
    <div class="van-actionsheet" style="display:none;">
      <ul class="van-hairline--bottom">
        <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">选项</span>
          <!---->
        </li>
        <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">选项</span>
          <!---->
        </li>
        <li class="van-actionsheet__item van-hairline--top">
          <div class="van-loading van-loading--circular van-loading--black van-actionsheet__loading" style="width:20px;height:20px;"><span class="van-loading__spinner van-loading__spinner--circular"> <svg viewBox="25 25 50 50" class="van-loading__circular"><circle cx="50" cy="50" r="20" fill="none"></circle></svg></span></div>
        </li>
      </ul>
      <div class="van-actionsheet__content"></div>
    </div>
  </demo-block>
  <demo-block title="">
    <button class="van-button van-button--default van-button--normal">
      <!----><span class="van-button__text"></span></button>
    <div class="van-actionsheet" style="display:none;">
      <ul class="van-hairline--bottom">
        <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">选项</span>
          <!---->
        </li>
        <li class="van-actionsheet__item van-hairline--top"><span class="van-actionsheet__name">选项</span>
          <!---->
        </li>
        <li class="van-actionsheet__item van-hairline--top">
          <div class="van-loading van-loading--circular van-loading--black van-actionsheet__loading" style="width:20px;height:20px;"><span class="van-loading__spinner van-loading__spinner--circular"> <svg viewBox="25 25 50 50" class="van-loading__circular"><circle cx="50" cy="50" r="20" fill="none"></circle></svg></span></div>
        </li>
      </ul>
      <div class="van-actionsheet__cancel van-hairline--top">取消</div>
    </div>
  </demo-block>
  <demo-block title="">
    <button class="van-button van-button--default van-button--normal">
      <!----><span class="van-button__text"></span></button>
    <div class="van-actionsheet van-actionsheet--withtitle" style="display:none;">
      <div class="van-hairline--top-bottom van-actionsheet__header">
        <div>标题</div>
        <i class="van-icon van-icon-close">
          <!---->
        </i>
      </div>
      <div class="van-actionsheet__content">
        <p>内容</p>
      </div>
    </div>
  </demo-block>
</demo-section>
`;
